src/Compilers/Core/MSBuildTask/Vbc.cs

```diff
@@ -273,6 +273,11 @@ public override bool Execute()
                 return false;
             }
 
+            if (SkipCompilerExecution)
+            {
+                return !Log.HasLoggedErrors;
+            }
+
             MovePdbFileIfNecessary(OutputAssembly.ItemSpec);
```

Comment:
Perhaps if (!Skip) MovePdbFileIfNecessary(...); to avoid multiple return statements.